Toolchain/Patches/gcc/0001-Add-a-gcc-driver-for-SerenityOS.patch

```diff
@@ -120,11 +120,11 @@ index 0000000000000000000000000000000000000000..7ee040c395e7bd404aa1f2b112e9248c
 +/* Files that are linked before user code.
 +   The %s tells GCC to look for these files in the library directory. */
 +#undef STARTFILE_SPEC
-+#define STARTFILE_SPEC "%{!shared:crt0.o%s} crti.o%s %{shared: %{!fbuilding-libgcc:crt0_shared.o%s}} %{shared|static-pie|!no-pie:crtbeginS.o%s; :crtbegin.o%s}"
++#define STARTFILE_SPEC "%{!shared:crt0.o%s} %{shared: %{!fbuilding-libgcc:crt0_shared.o%s}} %{shared|static-pie|!no-pie:crtbeginS.o%s; :crtbegin.o%s}"
 +
 +/* Files that are linked after user code. */
 +#undef ENDFILE_SPEC
-+#define ENDFILE_SPEC "%{shared|static-pie|!no-pie:crtendS.o%s; :crtend.o%s} crtn.o%s"
++#define ENDFILE_SPEC "%{shared|static-pie|!no-pie:crtendS.o%s; :crtend.o%s}"
```

Comment:
If you remove crti.o/crtn.o from the specs, I think you should also remove crtbegin{S,}.o and crtend{S,}.o?

Reply:
> I always thought crtbegin.o/crtend.o are only supposed to be used with the whole crti+crtn/.init+.fini thing.

FWIW, on my host, `crtbegin.o` seems to include the `[INIT,FINI]_ARRAY`.

> Though we also defined our own one in crt0_shared.cpp.

Are there some shenanigans that I don't know about with `weak` linkage? It seems to only be a _declaration_ to me, and thus is useless in a cpp file.

Reply:
> __dso_handle is a variable, so this should have an effect?

That's embarrassing :man_facepalming:, sorry about that